compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java

```diff
@@ -370,4 +395,179 @@ private BSymbol getFilterRequestFuncSymbol(BType filterType) {
+        if (isHttpResource(resourceNode)) {
+            addOrderParamConfig(resourceNode, env);
+        }
+    }
+
+    private void addOrderParamConfig(BLangFunction resourceNode, SymbolEnv env) {
+        List<BLangRecordLiteral.BLangRecordKeyValue> annotationValues = null;
+        for (BLangAnnotationAttachment annotationAttachment : resourceNode.getAnnotationAttachments()) {
+            if (ANN_RESOURCE_CONFIG.equals(annotationAttachment.getAnnotationName().getValue()) &&
+                    annotationAttachment.getExpression() != null) {
+                annotationValues = ((BLangRecordLiteral) annotationAttachment.getExpression()).keyValuePairs;
+                break;
+            }
+        }
+        if (annotationValues == null) {
+            return;
+        }
+        for (BLangRecordLiteral.BLangRecordKeyValue keyValue : annotationValues) {
+            switch (getAnnotationFieldKey(keyValue)) {
+                case ANN_RESOURCE_ATTR_WS_UPGRADE:
+                    for (BLangRecordLiteral.BLangRecordKeyValue upgradeField :
+                            ((BLangRecordLiteral) keyValue.getValue()).getKeyValuePairs()) {
+                        if (getAnnotationFieldKey(upgradeField).equals(ANN_RESOURCE_ATTR_WS_UPGRADE_PATH)) {
+                            addParamOrderConfigAnnotation(resourceNode, upgradeField.getValue(), env);
+                            break;
+                        }
+                    }
+                    break;
+                case ANN_RESOURCE_ATTR_PATH:
+                    addParamOrderConfigAnnotation(resourceNode, keyValue.getValue(), env);
+                    break;
+            }
+        }
+    }
+
+    private static String getAnnotationFieldKey(BLangRecordLiteral.BLangRecordKeyValue keyValue) {
+        return ((BLangSimpleVarRef) (keyValue.key).expr).variableName.getValue();
+    }
+
+    private static boolean checkForPathParam(List<BLangSimpleVariable> parameters, BLangExpression value) {
+        return parameters.size() > 2 && value.toString().contains("{") && value.toString().contains("}");
+    }
+
+    private void addParamOrderConfigAnnotation(BLangFunction resourceNode, BLangExpression value, SymbolEnv env) {
+        if (!checkForPathParam(resourceNode.getParameters(), value)) {
+            return;
+        }
+        DiagnosticPos pos = resourceNode.pos;
+        BLangAnnotationAttachment annoAttachment = (BLangAnnotationAttachment) TreeBuilder.createAnnotAttachmentNode();
+        resourceNode.addAnnotationAttachment(annoAttachment);
+        BSymbol annSymbol = lookupSymbolInPackage(symResolver, resourceNode.pos, env, names.fromString
+                (PACKAGE_NAME), names.fromString(ANN_RESOURCE_PARAM_ORDER_CONFIG), SymTag.ANNOTATION);
+
+        if (annSymbol instanceof BAnnotationSymbol) {
+            annoAttachment.annotationSymbol = (BAnnotationSymbol) annSymbol;
+        }
+        IdentifierNode identifierNode = TreeBuilder.createIdentifierNode();
```

Comment:
Can we replace 462-465 since 

```suggestion
        annoAttachment.annotationName = (BLangIdentifier) TreeBuilder.createIdentifierNode();;
```

TreeBuilder.createIdentifierNode() creates a BLangIdentifier right?